src/names.jl

```diff
@@ -112,7 +112,7 @@ arn_match(s, n, p) = occursin(p, s) ||
                      (debug_level == 0 || @warn("Bad ARN $n: \"$s\""); false)
 
 is_arn_prefix(s) = arn_match(s, "prefix",   r"^arn$")
-is_partition(s)  = arn_match(s, "partiton", r"^aws[a-z-]*$")
+is_partition(s)  = arn_match(s, "partition", r"^aws[a-z-]*$")
```

Comment:
This seems like a possibly major bug. Are there tests for this now?

Reply:
The usage of `partition` here is only in a `@warn` on line `112` above.

I'll write tests for these in `test/arn.jl`

Reply:
Ok, I'd suggest opening an issue and you can leave this as it is for this PR